### PR TITLE
Upgrades

### PR DESCRIPTION
**Pull request by danielui**
- set_default_partition method is more generic, user have to pass DevicesVersionPaylod as argument
- Made version_utils methods more usable, they returns dataclasses now
- Add get/set software install timeouts
- Moved upload image method to RepositoryAPI
- Add delete image method to RepositoryAPI

### CODE REVIEW

File: vmngclient/api/partition_manager_api.py [hunk 1]
```diff
@@ -48,21 +48,21 @@ def _set_default_partition(self, payload_devices: List[dict]) -> str:
...
+                going to be set
 
         Returns:
-            str: action id
+            str: set default partition action id
         """
-        devs = self.device_versions.get_devices_current_version(devices)
-        return self._set_default_partition(devs)
+        devices_current_versions = self.device_versions.get_devices_current_version(devices)
+        return self._set_default_partition(devices_current_versions)
```
Comment (igosoft) [inline vmngclient/api/partition_manager_api.py:61]:
```suggestion
        devices_versions = self.device_versions.get_devices_current_version(devices)
        return self._set_default_partition(devices_versions)
```

File: vmngclient/api/partition_manager_api.py [hunk 2]
```diff
@@ -72,12 +72,12 @@ def set_default_partition(self, devices: List[Device], version: str) -> str:
         Returns:
             str: action id
         """
-        devs = self.device_versions.get_device_list_in_installed(version, devices)
-        return self._set_default_partition(devs)
+        devices_choosen_version = self.device_versions.get_device_list_in_installed(version, devices)
+        return self._set_default_partition(devices_choosen_version)
```
Comment (igosoft) [inline vmngclient/api/partition_manager_api.py:76]:
```suggestion
        devices_version = self.device_versions.get_device_list_in_installed(version, devices)
        return self._set_default_partition(devices_version)
```

File: vmngclient/api/partition_manager_api.py [hunk 3]
```diff
@@ -48,21 +48,21 @@ def _set_default_partition(self, payload_devices: List[dict]) -> str:
...
+            devices (List[Device]): For those devices default partition is
+                going to be set
 
         Returns:
-            str: action id
+            str: set default partition action id
         """
-        devs = self.device_versions.get_devices_current_version(devices)
-        return self._set_default_partition(devs)
+        devices_current_versions = self.device_versions.get_devices_current_version(devices)
```
Comment (igosoft) [inline vmngclient/api/partition_manager_api.py:60]: it's API goldening . Do we really need it ?

File: vmngclient/api/partition_manager_api.py [hunk 4]
```diff
@@ -91,26 +91,45 @@ def remove_partition(self, devices: List[Device], version: str, force: bool = Fa
...
+        Returns:
+            str: remove action id
+        """
+        url = "/dataservice/device/action/removepartition"
+        payload = {
+            "action": "removepartition",
+            "devices": self.device_versions.get_devices_available_versions(devices),
+            "deviceType": "vmanage",
+        }
+        remove_action: Dict[str, str] = self.session.post(url, json=payload).json()
```
Comment (igosoft) [inline vmngclient/api/partition_manager_api.py:123]: also goldening. Following method will do a job 

def get_available_partitions()

then user will use get_available_partitions and remove_partitions

File: vmngclient/api/versions_utils.py [hunk 5]
```diff
@@ -138,23 +147,27 @@ def get_device_list_in_installed(self, version_to_set_up: str, devices: List[Dev
         """
         return self._get_device_list_in(version_to_set_up, devices, "installed_versions")
 
-    def get_device_list_in_available(self, version_to_set_up: str, devices: List[Device]) -> List[dict]:
+    def get_device_list_in_available(
```
Comment (igosoft) [inline vmngclient/api/versions_utils.py:150]:
```suggestion
    def get_device_available(
```

is should be good enough too :)

File: vmngclient/api/partition_manager_api.py (hunk 2 above)
Comment (danielui) [reply vmngclient/api/partition_manager_api.py]: Done

File: vmngclient/api/versions_utils.py (hunk 5 above)
Comment (danielui) [reply vmngclient/api/versions_utils.py]: Sure, done :)

File: vmngclient/api/partition_manager_api.py (hunk 3 above)
Comment (danielui) [reply vmngclient/api/partition_manager_api.py]: I changed it, from now we have one common method for set default partition.

File: vmngclient/api/partition_manager_api.py (hunk 4 above)
Comment (danielui) [reply vmngclient/api/partition_manager_api.py]: Same here, I removed that, and one common method is used to remove partitions.

File: vmngclient/api/software_action_api.py [hunk 6]
```diff
@@ -71,30 +69,31 @@ class SoftwareActionAPI:
     session = create_vManageSession(...)
 
     # Prepare devices list
-    devices = [dev for dev in DevicesAPI(session).devices if dev.hostname in ["vm5", "vm6"]]
-    software_image = PurePath("c8000v-universalk9.17.06.03a.0.56.SPA.bin")
+    devices = [device for device in DevicesAPI(session).devices
+                if dev.personality == Personality.VSMART]
```
Comment (tomaszwarda) [inline vmngclient/api/software_action_api.py:73]:
```suggestion
    devices = [device for device in DevicesAPI(session).devices
                if device .personality == Personality.VSMART]
```

File: vmngclient/api/versions_utils.py [hunk 7]
```diff
@@ -93,20 +110,72 @@ def get_image_version(self, software_image: str) -> Union[str, None]:
...
+    def _create_callback(self, encoder: MultipartEncoder):
+
+        bar = ProgressBar(expected_size=encoder._calculate_length(), filled_char="=")
+
+        def callback(monitor: MultipartEncoderMonitor):
+            bar.show(monitor.bytes_read)
+
+        return callback
+
+    def upload_image(self, image_path: str) -> int:
```
Comment (tomaszwarda) [inline vmngclient/api/versions_utils.py:122]: Maybe example of upload and delete image in doc sting?

Comment (danielui) [reply vmngclient/api/versions_utils.py:122]: I think, that those 2 methods are too simple in use, to put usage example of them into doc string.

**Review by tomaszwarda — APPROVED** [review]

**Review by sbasan — APPROVED** [review]
lgtm

**Pull request merged**